monkey/monkey_island/cc/resources/auth/register_agent.py

```diff
@@ -0,0 +1,38 @@
+import json
+
+from flask import make_response, request
+
+from monkey_island.cc.flask_utils import AbstractResource
+from monkey_island.cc.server_utils.response_utils import response_to_invalid_request
+
+
+class RegisterAgent(AbstractResource):
+    """
+    A resource for registering an Agent
```

Comment:
This isn't for registering an agent at all. It's for authenticating with a OTP. OTP bearers may or may not be agents. This endpoint shouldn't make assumptions about the caller.

Reply:
Maybe `/api/otp-login` would be ok?